File: solutions_1483485_0/C++/nking/A-small-0.cpp
#include <iostream>
#include <string>
#include <vector>
#include <list>
#include <map>
#include <algorithm>
#include <cstdlib>
#include <cstdio>
#include<cstring>

using namespace std;
int main() {
	char mapper[26];
	string sample[3];
	sample[0] = "ejp mysljylc kd kxveddknmc re jsicpdrysi";
	sample[1] = "rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd";
	sample[2] = "de kr kd eoya kw aej tysr re ujdr lkgc jv";

	string mapping[3];
	mapping[0] = "our language is impossible to understand";
	mapping[1] = "there are twenty six factorial possibilities";
	mapping[2] = "so it is okay if you want to just give up";

	for (int i = 0; i < 3; i++) {
		for (int j = 0; j < sample[i].length(); ++j) {
			if (sample[i].at(j) != ' ') {
				mapper[sample[i].at(j) - 'a'] = mapping[i].at(j);
			}
		}
	}
	mapper['y' - 'a'] = 'a';
	mapper['q' - 'a'] = 'z';
	mapper['e' - 'a'] = 'o';
	mapper['z' - 'a'] = 'q';
	int T;
	cin >> T;
	char s[110];
	char m[110];
	cin.getline(s, 105);
	for (int i = 1; i <= T; ++i) {
		cin.getline(s, 105);
		for (int j = 0; j <= strlen(s); ++j) {
			if (isalpha(s[j])) {
				m[j] = mapper[s[j] - 'a'];
			} else {
				m[j] = s[j];
			}
		}
		cout << "Case #" << i << ": " << m << endl;
	}
	return 0;
}

File: solutions_1483485_0/Python/willo/A-small-0.py
from operator import itemgetter, attrgetter
import itertools
import os, logging

FILE = 'A-small-0'

def runner():
	#path = os.path.join(os.path.dirname(__file__), 'input', '%s.in' % FILE)
	path = os.path.join(os.path.dirname(__file__), '%s.in' % FILE)
	f = open(path, 'r')
	
	path = os.path.join(os.path.dirname(__file__), '%s.out' % FILE)
	r = open(path, 'w')
	
	for n, test_case in enumerate(f.readlines()[1:]):
#		if n % 2 == 1:

		result = encode(test_case.strip())

#		if result == None: result = 'NO'
		s = 'Case #%r: %s\n' % (n + 1, result)
		print s
		r.write(s)
	
	f.close()
	r.close()
	

def unencode(l):
    for t in l:
        for n, c in enumerate(t[0]):
            if c != ' ':
                MAP[t[1][n]] = c

#MAP = {}
#unencode([('ejp mysljylc kd kxveddknmc re jsicpdrysi', 'our language is impossible to understand'), ('rbcpc ypc rtcsra dkh wyfrepkym veddknkmkrkcd', 'there are twenty six factorial possibilities'), ('de kr kd eoya kw aej tysr re ujdr lkgc jv', 'so it is okay if you want to just give up'), ('y qee', 'a zoo')])

MAP = {'a': 'y', 'c': 'f', 'b': 'n', 'e': 'c', 'd': 'i', 'g': 'l', 'f': 'w', 'i': 'k', 'h': 'b', 'k': 'o', 'j': 'u', 'm': 'x', 'l': 'm', 'o': 'e', 'n': 's', 'p': 'v', 's': 'd', 'r': 'p', 'u': 'j', 't': 'r', 'w': 't', 'v': 'g', 'y': 'a', 'x': 'h', 'z': 'q'}
MAP['q'] = 'z'
MAP[' '] = ' '

UNMAP = {v: k for k, v in MAP.items()}

def encode(s):
    '''
    >>> encode('a zoo')
    'y qee'
    '''
    r = ''
    for c in s:
        r += UNMAP[c]

    return r


    

if __name__ == "__main__":
	#import doctest
	#doctest.testmod()
	runner()

